lib/internal/async_hooks.js

```diff
@@ -103,6 +105,26 @@ const emitDestroyNative = emitHookFactory(destroy_symbol, 'emitDestroyNative');
 const emitPromiseResolveNative =
     emitHookFactory(promise_resolve_symbol, 'emitPromiseResolveNative');
 
+function callbackTrampoline(asyncId, cb, domain_cb, ...args) {
+  if (hasHooks(kBefore))
+    emitBeforeNative(asyncId);
+
+  let result;
+  if (typeof domain_cb === 'function') {
+    args.unshift(cb);
```

Comment:
Fwiw, this could also be using the primordial:

```suggestion
    ArrayPrototypeUnshift(args, cb);
```

Reply:
Oh, good point. I'll do that!